landoapi/models/landing_job.py

```diff
@@ -248,11 +248,7 @@ def job_queue_query(
             grace_cutoff = now - datetime.timedelta(seconds=grace_seconds)
             q = q.filter(cls.created_at < grace_cutoff)
 
-        # Any `LandingJobStatus.IN_PROGRESS` job is first and there should
-        # be a maximum of one (per repository). For
-        # `LandingJobStatus.SUBMITTED` jobs, higher priority items come first
-        # and then we order by creation time (older first).
-        q = q.order_by(cls.status.desc(), cls.priority.desc(), cls.created_at)
+        q = q.order_by(cls.priority.desc(), cls.status.desc(), cls.created_at)
```

Comment:
Do we need this change? We already get the highest priority items with the highest priority statuses, right?